client/a8c-for-agencies/hooks/use-track-provisioning-sites.ts

```diff
@@ -0,0 +1,67 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const TTL_DURATION = 300000; // 5 minutes in miliseconds
+
+type ProvisioningSite = {
+	id: number;
+	migration?: boolean;
+	development?: boolean;
+	ttl: number;
+};
+
+export default function useTrackProvisioningSites() {
+	const [ provisioningSites, setProvisioningSites ] = useState< ProvisioningSite[] >( [] );
+
+	const reload = useCallback( () => {
+		const storedSites = localStorage.getItem( 'provisioningSites' );
+		if ( storedSites ) {
+			const sites: ProvisioningSite[] = JSON.parse( storedSites );
+			// We remove the site from tracking if it is more than 5 minutes.
+			const activeSites = sites.filter( ( { ttl } ) => ttl >= new Date().getTime() );
+
+			setProvisioningSites( activeSites );
+			localStorage.setItem( 'provisioningSites', JSON.stringify( activeSites ) );
+		}
+	}, [] );
+
+	const trackSiteId = useCallback(
+		(
+			siteId: number,
+			{ migration, development }: { migration?: boolean; development?: boolean } = {}
+		) => {
+			const updatedSites = [
+				...provisioningSites,
+				{ id: siteId, migration, development, ttl: new Date().getTime() + TTL_DURATION },
+			];
+
+			localStorage.setItem( 'provisioningSites', JSON.stringify( updatedSites ) );
+			window.dispatchEvent( new CustomEvent( 'a4a-provisioning-site-change' ) );
```

Comment:
What do you think about moving this event to a constant like PROVISIONING_SITE_CHANGE_EVENT?

Reply:
Good point! this has been addressed. 👍